influxql/call_iterator.go

```diff
@@ -362,18 +360,30 @@ func booleanLastReduce(prev, curr *BooleanPoint) (int64, bool, []interface{}) {
 func NewDistinctIterator(input Iterator, opt IteratorOptions) (Iterator, error) {
 	switch input := input.(type) {
 	case FloatIterator:
-		return &floatReduceSliceIterator{input: newBufFloatIterator(input), opt: opt, fn: floatDistinctReduceSlice}, nil
+		createFn := func() (FloatPointSliceAggregator, FloatPointSliceEmitter) {
```

Comment:
The aggregator and emitter should work for individual points and emitters. If you make the emitter return a slice instead of a single point, that interface will work for all aggregates and the iterator types can be reused. I can clarify this if needed or help with the change.